### PR TITLE
sovereign flag

### PR DESCRIPTION
**Pull request by axenteoctavian**

### CODE REVIEW

File: cmd/elasticindexer/flags.go [hunk 1]
```diff
@@ -47,4 +47,9 @@ var (
 		Name:  "disable-ansi-color",
 		Usage: "Boolean option for disabling ANSI colors in the logging system.",
 	}
+	// sovereign defines a flag that specifies if what run type components should use
```
Comment (mariusmihaic) [inline cmd/elasticindexer/flags.go:50]:
```suggestion
	// sovereign defines a flag that specifies if the es instance should run for a sovereign chain
```

File: factory/wsIndexerFactory.go [hunk 2]
```diff
@@ -7,23 +7,24 @@ import (
...
 	"github.com/multiversx/mx-chain-es-indexer-go/process/factory"
 	"github.com/multiversx/mx-chain-es-indexer-go/process/wsindexer"
-	logger "github.com/multiversx/mx-chain-logger-go"
 )
 
 var log = logger.GetOrCreate("elasticindexer")
 
 // CreateWsIndexer will create a new instance of wsindexer.WSClient
-func CreateWsIndexer(cfg config.Config, clusterCfg config.ClusterConfig, statusMetrics core.StatusMetricsHandler, version string) (wsindexer.WSClient, error) {
+func CreateWsIndexer(cfg config.Config, clusterCfg config.ClusterConfig, statusMetrics core.StatusMetricsHandler, version string, isSovereignType bool) (wsindexer.WSClient, error) {
```
Comment (mariusmihaic) [inline factory/wsIndexerFactory.go:21]: `isSovereignType` should be somewhere loaded from configs, not added as a new bool variable.

File: process/factory/indexerFactory.go [hunk 3]
```diff
@@ -53,6 +57,15 @@ func NewIndexer(args ArgsIndexerFactory) (dataindexer.Indexer, error) {
 		return nil, err
 	}
 
+	if args.SovereignConfig {
```
Comment (mariusmihaic) [inline process/factory/indexerFactory.go:60]: No if/else here, please.
RunTypeComp creation should be at the highest level (main file) and passed down.

**Review by mariusmihaic — CHANGES_REQUESTED** [review]

File: process/factory/indexerFactory.go (hunk 3 above)
Comment (axenteoctavian) [reply process/factory/indexerFactory.go]: I had to create runType components here because here is where the node is connecting with indexer factory

**Review by sasurobert — DISMISSED** [review]

**Review by miiu96 — DISMISSED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**